tuf/api/metadata.py

```diff
@@ -1274,12 +1332,15 @@ class DelegatedRole(Role):
     instance attributes.*
 
     Args:
-        name: Delegated role name.
+        name: Delegated role name. If succinct_hash_info is passed, then a None
+            value should be given to ``name``.
```

Comment:
Note that there is quite a bit of text on the mutual exclusivity of arguments above this: probably makes sens to mention this name issue there and fix that text in general.